src/integration.md

```diff
@@ -0,0 +1,569 @@
+tag-fem:
+
+# Integration
+
+MFEM's spatial integrations are performed in the usual finite element
+manner by first splitting the spatial domain into a collection of
+non-overlapping "elements" which cover the domain. This is usually
+referred to as the "mesh". An integral can then be computed separately
+in each element and the results added together:
+
+$$ \int_\Omega f(x)\,d\Omega = \sum_i\int_{\Omega_i}f(x)\,d\Omega $$
+
+Where $\Omega$ is the full domain and $\Omega_i$ is the domain of the
+i-th element. In MFEM this sum over elements is performed in classes
+such as the `BilinearForm` or `LinearForm` and their parallel
+counterparts.
+
+Elements come in a variety of shapes and they may be flat-sided or
+curved. For this reason it is much simpler to perform the element-wise
+integrations on reference elements which have relatively simple
+shapes. For example in 2D we might integrate over a unit square rather
+than an arbitrary quadrilateral.
+
+Finite element methods typically make the assumption that the
+functions to be integrated are non-singular and at least reasonably
+smooth. This enables us to employ families of relatively simple
+quadrature rules which are designed for accurately integrating
+polynomials.  This is in contrast to boundary element methods which
+require more specialized rules which can accurately integrate
+singularities. Our rules take the form:
+
+$$\int_{\Omega_i} f(x)\,d\Omega \approx \sum_j w_j\,f(x(u_j))\,|J_i(u_j)|\label{eq:quad_rule}$$
+
+Where $w_j$ are the quadrature weights, $u_j$ are the quadrature
+points within the reference element, and $|J_i(u_j)|$ is the Jacobian
+determinant for element $i$ at the location $u_j$. Integrals at this level
+are typically computed by classes derived from
+`BilinearFormIntegrator` or `LinearFormIntegrator`, see [Bilinear Form
+Integrators](bilininteg.md) or [Linear Form Integrators](lininteg.md)
+for numerous examples.
+
+## Integration Rules
+
+The basic building block of an integration rule is the
+`IntegrationPoint`. This is a minimal object with member data 'x',
+'y', 'z', and 'weight' (and an integer 'index' which is a mystery to
+me). These store the coordinates of the integration point in the
+reference coordinate system, $u_j$ from equation $\ref{eq:quad_rule}$
+is defined as $u_j\equiv(x,y,z)$ , along with the quadrature weight,
+$w_j$ also from equation $\ref{eq:quad_rule}$.
+
+Integration points can be collected together into an `IntegrationRule`
+object. `IntegrationRule` is little more than a container for the set
+of `IntegrationPoint` objects associated with an integration rule for
+a given order of accuracy within the domain of a specific reference
+element.
+
+`IntegrationRule` objects are in turn collected together into the
+`IntegrationRules` global object. This object constructs and caches
```

Comment:
I know it's a bit pedantic, but I think it may avoid confusion.  IntegrationRules is the name of the class whereas IntRules is the name of the global object that people will be using.  Generally class names are good adjectives, but poor nouns.